socs/db/suprsync_cli.py

```diff
@@ -41,17 +41,30 @@ def list_func(args):
         print(f"{i}: {f.local_path}")
 
 
-if __name__ == '__main__':
+def add_local_files_func(args):
+    srfm = SupRsyncFilesManager(args.db, create_all=args.create_db)
+    known_files = srfm.get_known_files(args.archive_name)
+    known_paths = [f.local_path for f in known_files]
+    for root, _, files in os.walk(args.local_root):
+        for file in files:
+            path = os.path.join(root, file)
+            path = os.path.abspath(path)
+            if path not in known_paths:
+                remote_path = os.path.relpath(path, args.local_root)
+                srfm.add_file(path, remote_path, args.archive_name)
```

Comment:
Is there any checking on local_root, to ensure that it's the same local root used in the past with this DB?

For codes that work with the filesystem and a database, it's helpful to have a --dry-run switch -- which would cause the code to print out what it _would do_ without actually doing it.  In this case it would allow one to see all the files the config thinks are new; a good way to catch usage mistakes during initial setup (such as passing local_root="/path/to" instead of local_root="/path/to/stuff").

Reply:
Good idea, I added a `--dry` argument which will just print out some debugging info and up to 10 files that will be added to the database